app/Http/Controllers/SearchController.php

```diff
@@ -85,7 +86,13 @@ public function search(Request $request)
 		 * Photos
 		 */
 		// for now we only look in OUR pictures
-		$query = Photo::where('owner_id', '=', $id);
+		$query = Photo::where('id', '>', 0);
+		$query->where(function (Builder $query) use ($id) {
+			$query->where('owner_id', '=', $id);
+			if (Configs::get_value('public_search', '0') == '1') {
+				$query = $query->orWhere('public', '>=', 0);
+			}		
+		});
```

Comment:
This is not the right way to do it -- but I understand where you're coming from, since the original code was taking shortcuts so it wasn't correct either (hence the `for now we only look in OUR pictures` comment). I can think of at least two problems here:

1. Checking the `public` flag of a `Photo` is insufficient. For photos in albums, it's actually the `public` flag of the _album_ that determines if an image is public or not.

2. This will ignore shared albums (albums that another user is sharing privately with the current user). But this problem was there before your changes as well.

I'll say more about how I think this _should_ be done below, when talking about Albums.

Reply:
> 1. Checking the `public` flag of a `Photo` is insufficient. For photos in albums, it's actually the `public` flag of the _album_ that determines if an image is public or not.

And if it was sufficient, it would even beimplemented wrong with `public >= 0`. This was just another trick for the query generator to first fetch all photos. Non-public images are not getting into the response object, see your next comment: https://github.com/LycheeOrg/Lychee-Laravel/pull/256/files#r285669443